clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/op/handler/KubernetesHandler.java

```diff
@@ -51,6 +57,19 @@
   abstract public SpinnakerKind spinnakerKind();
   abstract public Status status(KubernetesManifest manifest);
 
+  public ManifestBasedModel fromCacheData(KubernetesV2CacheData cacheData) {
```

Comment:
Should this be `static`?

Reply:
Ah I guess not because you plan to override it? It's a little strange that there's a non-static instance method that handles every type of cache data - but will be overridden to handle only one type of cache data...

Reply:
If you had a `KubernetesServerGroupHandler` that implements this method and does only the server group bit, and is implemented by replica set, stateful set, daemon set, etc... and the same for server group managers, I think the pattern would be a little clearer.

Reply:
Hmm... I don't think that would work for custom resources. The problem is that the default mapping has to be encoded with the `SpinnakerKind`.